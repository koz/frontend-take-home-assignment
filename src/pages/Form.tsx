import React from 'react';
import styled from 'styled-components';
import { media } from '../styles/mediaQueries';
import homeSVG from '../icons/home.svg';

import SavingCard from '../components/SavingCard';
import PageLayout from '../components/PageLayout';

const StyledTitle = styled.h1`
  color: #1b31a8;
  font-size: 1.8rem;
  line-height: 1.44;
  margin-top: 2.3rem;
  margin-bottom: 2.4rem;

  ${media.desktop} {
    font-size: 2.4rem;
    line-height: 1.33;
    margin-top: 5.4rem;
    margin-bottom: 4rem;
  }

  & > strong {
    font-weight: 600;
  }
`;

const StyledSavingCard = styled(SavingCard)`
  align-self: stretch;

  ${media.mobile} {
    flex-grow: 1;
  }

  ${media.desktop} {
    align-self: center;
    width: 56rem;
  }
`;

const Form = () => (
  <PageLayout>
    <StyledTitle>
      Let&apos;s plan your <strong>saving goal</strong>
    </StyledTitle>
    <StyledSavingCard
      iconSrc={homeSVG}
      iconAlt="Little house with dollar sign on the bottom left"
      title="Buy a house"
    />
  </PageLayout>
);

export default Form;
