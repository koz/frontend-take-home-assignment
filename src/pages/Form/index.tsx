import React from 'react';
import styled from 'styled-components';
import { media } from '../../styles/mediaQueries';
import homeSVG from '../../icons/home.svg';

import SavingCard from '../../components/SavingCard';
import PageLayout from '../../components/PageLayout';
import { useParams } from 'react-router-dom';

const StyledTitle = styled.h1`
  color: #1b31a8;
  font-size: 1.8rem;
  line-height: 1.44;
  margin-top: 2.3rem;
  margin-bottom: 2.4rem;

  ${media.desktop} {
    font-size: 2.4rem;
    line-height: 1.33;
    margin-top: 5.4rem;
    margin-bottom: 4rem;
  }

  & > strong {
    font-weight: 600;
  }
`;

const StyledSavingCard = styled(SavingCard)`
  align-self: stretch;

  ${media.mobile} {
    flex-grow: 1;
  }

  ${media.desktop} {
    align-self: center;
    width: 56rem;
  }
`;

const Form = () => {
  const { id } = useParams();
  return (
    <PageLayout>
      <StyledTitle>
        Let&apos;s plan your <strong>saving goal</strong>
      </StyledTitle>
      <StyledSavingCard
        iconSrc={homeSVG}
        iconAlt="Little house with dollar sign on the bottom left"
        title="Buy a house"
      />
    </PageLayout>
  );
};

export default Form;
