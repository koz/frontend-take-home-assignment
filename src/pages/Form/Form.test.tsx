import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Form from './index';

jest.mock('react-router-dom', () => ({
  useParams: jest.fn().mockReturnValue({ id: 'car' })
}));

describe('<Form />', () => {
  it('matches snapshot', () => {
    const component = shallow(<Form />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
