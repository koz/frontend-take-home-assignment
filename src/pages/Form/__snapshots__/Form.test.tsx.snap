// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> matches snapshot 1`] = `
<PageLayout>
  <styled.h1>
    Let's plan your 
    <strong>
      saving goal
    </strong>
  </styled.h1>
  <Styled(SavingCard)
    iconAlt="Icon alt text"
    iconSrc={Object {}}
    id="car"
    title="Buy a car"
  />
</PageLayout>
`;
